code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,197 @@
+/client/proc/cmd_select_equipment(mob/M in GLOB.mob_list)
+	set category = "Admin.Events"
+	set name = "Select equipment but better"
+
+
+	var/datum/select_equipment/ui  = new(usr, M)
+	ui.ui_interact(usr)
+
+/datum/select_equipment
+	var/client/user
+	var/mob/target
+
+	var/dummy_key
+	var/mob/living/carbon/human/dummy/dummy
+
+	var/static/list/cached_outfits
+
+	//normally a path; an initialized outfit object if it's a custom outfit
+	var/datum/outfit/selected_outfit = /datum/outfit
+	//used to keep track of which outfit the UI has selected
+	var/selected_name = "/datum/outfit"
+
+/datum/select_equipment/New(usr, mob/M)
+	user = CLIENT_FROM_VAR(usr)
+
+	if(!(ishuman(M) || isobserver(M)))
+		alert("Invalid mob")
+		return
+	target = M
+
+/datum/select_equipment/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SelectEquipment", "Select Equipment")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/select_equipment/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/select_equipment/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target))
+		return UI_CLOSE
+	return ..()
+
+/datum/select_equipment/ui_close(mob/user)
+	clear_human_dummy(dummy_key)
+	qdel(src)
+
+/datum/select_equipment/proc/init_dummy()
+	dummy_key = "selectequipmentUI_[target]"
+	dummy = generate_or_wait_for_human_dummy(dummy_key)
+	var/mob/living/carbon/C = target
+	if(istype(C))
+		C.dna.transfer_identity(dummy)
+		dummy.updateappearance()
+
+	unset_busy_human_dummy(dummy_key)
+	return
+
+/*
+ * category (string) - The tab it will be under
+ * path (typepath or string) - This will sent this back to ui_act to preview or spawn in an outfit
+ * name (string) - Will be the text on the button
+ * priority (int) - default 0, 1 for favorites, 2 for priority buttons
+*/
+#define OUTFIT_ENTRY(category, path, name, priority) list("category" = category, "path" = path, "name" = name, "priority" = priority)
+//passes a ref instead of path, every entry needs some sort of unique identifier
+//and GLOB.custom_outfits allows duplicate names
+#define OUTFIT_CUSTOM_ENTRY(category, ref, name, priority) list("category" = category, "ref" = ref, "name" = name, "priority" = priority)
+
+/datum/select_equipment/proc/make_outfit_entries(category="General", list/L)
+	var/list/entries = list()
+	for(var/path in L)
+		var/datum/outfit/O = path
+		var/priority = 0
+		if(path == /datum/outfit/job/roboticist) //dummy check here until I get favorites working
+			priority = 1
+		entries += list(OUTFIT_ENTRY(category, path, initial(O.name), priority))
+	return entries
+
+
+//GLOB.custom_outfits lists outfit *objects* so we'll need to do some custom handling for it
+/datum/select_equipment/proc/make_custom_outfit_entries(list/L)
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names